x-pack/plugins/maps/public/components/toolbar_overlay/view.js

```diff
@@ -0,0 +1,284 @@
+      drawBoundsAction.onClick = () => this._activateDrawForFirstIndexPattern(DRAW_TYPE.BOUNDS);
+    } else {
+      drawPolygonAction.panel = this._getIndexPatternSelectionPanel(1);
+      drawPolygonAction.onClick = () => this._showIndexPatternSelection(DRAW_TYPE.POLYGON);
+      drawBoundsAction.panel = this._getIndexPatternSelectionPanel(2);
+      drawBoundsAction.onClick = () => this._showIndexPatternSelection(DRAW_TYPE.BOUNDS);
+    }
+
+    return flattenPanelTree({
+      id: 0,
+      title: i18n.translate('xpack.maps.toolbarOverlay.tools.toolbarTitle', {
+        defaultMessage: 'Tools',
+      }),
+      items: [drawPolygonAction, drawBoundsAction]
+    });
+  }
+
+  _getIndexPatternSelectionPanel(id) {
+    const options = this.state.uniqueIndexPatternsAndGeoFields.map((indexPatternAndGeoField) => {
+      return {
+        label: `${indexPatternAndGeoField.indexPatternTitle} : ${indexPatternAndGeoField.geoField}`,
+        value: indexPatternAndGeoField
+      };
+    });
+
+    const renderGeoField = (option, searchValue) => {
+      return (
+        <Fragment>
+          <EuiTextColor color="subdued">
+            <small>
+              <EuiHighlight search={searchValue}>{option.value.indexPatternTitle}</EuiHighlight>
+            </small>
+          </EuiTextColor>
+          <br />
+          <EuiHighlight search={searchValue}>
+            {option.value.geoField}
+          </EuiHighlight>
+        </Fragment>
+      );
+    };
+
+    const indexPatternSelection = (
+      <EuiSelectable
+        searchable
+        searchProps={{
+          placeholder: 'Filter list',
+          compressed: true,
+        }}
+        options={options}
+        /**
+         * *TODO*: FIX this handler as EuiSelectable passes back the full options
+         * list with the selected option set with `checked: 'on'`
+         */
+        onChange={this._onIndexPatternSelection}
+        renderOption={renderGeoField}
+        listProps={{
+          rowHeight: 50,
+          showIcons: false,
+        }}
+      >
+        {(list, search) => (
+          <div>
+            {search}
+            {list}
+          </div>
+        )}
+      </EuiSelectable>
+    );
+
+    return {
+      id: id,
+      title: i18n.translate('xpack.maps.toolbarOverlay.geofield.toolbarTitle', {
+        defaultMessage: 'Select geo field',
+      }),
+      content: indexPatternSelection
+    };
+  }
+
+  _renderToolbarButton() {
+    return (
+      <EuiButtonIcon
+        className="mapToolbarOverlay__button"
+        color="text"
+        iconType="wrench"
+        onClick={this._openToolbar}
+        aria-label={i18n.translate('xpack.maps.toolbarOverlay.toolbarIconTitle', {
+          defaultMessage: 'Tools',
+        })}
+        title={i18n.translate('xpack.maps.toolbarOverlay.toolbarIconTitle', {
+          defaultMessage: 'Tools',
+        })}
+      />
+    );
+  }
+
+  render() {
+
+    if (
+      !this.props.isReadOnly ||
```

Comment:
Why does this component not render when `!this.props.isReadOnly`? Why does isReadOnly have any effect on toolbar?

Reply:
no need to show toolbar when there are no functional tools inside the toolbar. I would refactor this when we introduce additional tools that _are_ functional. Until then, this would just be a fluffy code-path.

Reply:
Not sure I follow. Shouldn't the check be for `isFilterable` instead?

Reply:
Ok. got it. +1